### PR TITLE
Update version to 2.14.0

### PR DESCRIPTION
**Pull request by VaishnaviHire**
## Description
https://github.com/opendatahub-io/opendatahub-community/issues/153

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by mattmahoneyrh — APPROVED** [review]

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -1725,7 +1724,7 @@ spec:
                 env:
                 - name: DISABLE_DSC_CONFIG
                   value: "true"
-                image: REPLACE_IMAGE:latest
+                image: quay.io/opendatahub/opendatahub-operator:latest
```
Comment (zdtsw) [inline bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:1727]: we wanna change it here?

Comment (VaishnaviHire) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:1727]: Yeah I will update it

**Review by zdtsw — APPROVED** [review]

**Pull request merged**